sdk/python/kfp/components/component_decorator.py

```diff
@@ -23,6 +23,7 @@ def component(func: Optional[Callable] = None,
               base_image: Optional[str] = None,
               target_image: Optional[str] = None,
               packages_to_install: List[str] = None,
+              pip_index_urls: Optional[List[str]] = None,
```

Comment:
Adding a parameter in the middle would be a breaking change for those who use positional arguments. We should probably add this at the end instead. WDYT?

Reply:
I think the `*` on line 22 makes all but `fun` a [keyword-only argument](https://peps.python.org/pep-3102/#specification). Is there an implementation I'm not thinking of?

Reply:
Ah, you're absolutely right. I missed that `*` above.

Reply:
How do you like the ordering of the params in general? Ideally it's also reasonable/intuitive for contributors to `kfp` :)